### PR TITLE
Support progressive JPEGs

### LINKED ISSUE #378: Support progressive JPEGs
https://www.wired.com/2013/01/the-return-of-the-progressive-jpeg/

### PR DESCRIPTION
**Pull request by philip-hunt**
Close #378

### CODE REVIEW

**Comment by eduardoboucas** [comment]
I was thinking the same thing. Also, if we don't do that, it would technically be a breaking change and some clients might not support/want progressive JPG?

File: dadi/lib/handlers/image.js [hunk 1]
```diff
@@ -1020,6 +1026,23 @@ ImageHandler.prototype.processGif = function (buffer) {
...
+ * @param {Buffer} buffer - a Buffer extracted from the main image
+ * processor after applying image manipulations
+ * @returns {Buffer} a progressive JPEG encoded buffer
+ */
+ImageHandler.prototype.processJpeg = function (buffer) {
+  return imagemin.buffer(buffer, {
+    plugins: [
+      imageminJpegtran({progressive: true})
+    ]
+  }).then(outputBuffer => {
```
Comment (eduardoboucas) [inline dadi/lib/handlers/image.js]: Do we need this `then`?

Comment (jimlambie) [reply dadi/lib/handlers/image.js]: Is there a header we can inspect to tell if a client supports progressive jpg? @abovedave submitted the issue so am deferring to him.

Comment (philip-hunt) [reply dadi/lib/handlers/image.js]: Got spot, I'll remove that

**Comment by philip-hunt** [comment]
We could either add a boolean 'progressive' argument for JPEGs or create compatibility module which can determine browser compatibility for different features. I had a look and there doesn't seem to be a test we can run for progressive other than parsing the UA string and checking it against a list of supported browser versions (looks like anything in the last 10/15 years has support).

**Comment by jimlambie** [comment]
@philip-hunt let's make progressive the default behaviour, but allow a url param to override it.

We can review afterwards to see if there's a performance hit.

**Pull request merged**